### DISCUSSION TITLE
What's the difference in shaders with ReShade?

### DISCUSSION THREAD

**Discussion by Aberro** [retroluxfilm/reshade-vrtoolkit#8] [Q&A] [answered]
As title says. How shaders for VRToolkit differs from shaders from ReShade? Is it possible to rewrite ReShade shaders for VRToolkit and, if so, how?

**Comment by retroluxfilm** [comment, marked as answer]
The VRTookit shader is optimized to reduce the performance impact in VR compared when you use each shader individually. This is done by merging common shaders (Sharpening shaders, Color correction shaders, etc.) into one single pass. With that each pixel will only be processed once per frame instead per activated shader. To further optimize performance it also masks out all pixels from shader processing that are not in the view or the speed spot of your headset. The circular mask can be configure it to match your headset field of view and offset for the wide hmds like the pimax.

In additon some of the merged shaders have minor adjustemnts, most notably the CAS shader have been optimized to produce less artifacts by masking out high contrast edges (user controllable).

If you want to extend or simply experiment with the vrtoolkit shader you can clone the repository and get all the files needed to start away. Its in the `Shader` folder
